src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,92 @@
-export const PeopleFilters = () => {
+import { Link, useSearchParams } from 'react-router-dom';
+import { Person } from '../types/Person';
+import React, { useState, useEffect } from 'react';
+
+interface PeopleFiltersProps {
+  persons: Person[];
+  onFilterChange: (filter: 'all' | 'male' | 'female') => void;
+  onCenturyChange: (century: string | 'all' | null) => void;
+  onSearchChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const PeopleFilters: React.FC<PeopleFiltersProps> = ({
+  onFilterChange,
+  onCenturyChange,
+  onSearchChange,
+}) => {
+  const [filter, setFilter] = useState<'all' | 'male' | 'female'>('all');
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [searchParams, setSearchParams] = useSearchParams();
+  const centuryQuery = searchParams.getAll('century') || [];
+
+  useEffect(() => {
+    const currentFilter = searchParams.get('filter') as
+      | 'all'
+      | 'male'
+      | 'female';
+    const currentSearch = searchParams.get('search') || '';
+
+    if (currentFilter) {
+      setFilter(currentFilter);
+    }
+
+    setSearchQuery(currentSearch);
+  }, [searchParams, centuryQuery]);
+
+  const handleAllClick = () => {
+    const newSearchParams = new URLSearchParams(searchParams);
+
+    newSearchParams.delete('century');
+    newSearchParams.set('filter', 'all');
+    setSearchParams(newSearchParams);
+    onCenturyChange(null);
+    onFilterChange('all');
+  };
+
+  const handleResetFilters = () => {
+    const newSearchParams = new URLSearchParams();
+
+    setSearchParams(newSearchParams);
+    onCenturyChange(null);
+    onFilterChange('all');
+    setSearchQuery('');
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">
+        <Link
+          className={filter === 'all' ? 'is-active' : ''}
+          to="#"
+          onClick={e => {
+            e.preventDefault();
+            onFilterChange('all');
```

Comment:
The `onFilterChange('all')` is already being called in the `handleAllClick` function. Consider using `handleAllClick` here to avoid code duplication.